api/v1alpha1/vaultauth_types.go

```diff
@@ -51,6 +51,15 @@ type VaultAuthConfigJWT struct {
 	TokenExpirationSeconds int64 `json:"tokenExpirationSeconds,omitempty"`
 }
 
+// VaultAuthConfigAppRole provides VaultAuth configuration options needed for authenticating to Vault.
+type VaultAuthConfigAppRole struct {
+	// RoleID (role_id) to use for authenticating to Vault.
+	RoleID string `json:"roleid"`
+	// Selects a key of a secret in the AuthMethod's namespace which holds the secret_id of the approle used
```

Comment:
I wonder if we might be more opinionated here and  require a specific key be present in the secret. We could at a minimum set a `default` value.

Reply:
Hrm, I suppose we could do this, I need to figure out how to use kubebuilder to define defaults for a custom object.
What do you envision that `key` would be?
I personally feel like a user is probably used to the concept of a secretRef's name+key convention enough to probably not forget the key when creating the reference?
Happy to make the change :)